src/core/verification.service.ts

```
@@ -150,12 +308,36 @@
 
       const proofInput = assertion.verification as OB3.Proof; // Keep as Proof initially for verificationMethod access
 
+      // Check if this is a JWT proof
+      if (isJWTProof(proofInput)) {
+        // Extract keyId from verification method if possible
+        let keyId = 'default';
+        if (proofInput.verificationMethod) {
+          try {
+            const VMethodStr = proofInput.verificationMethod as string;
+            const match = VMethodStr.match(/\/public-keys\/([^#\/]+)/);
+            if (match && match[1]) {
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)